src/transpose.jl

```diff
@@ -63,15 +49,8 @@ majortranspose(::FourthOrderTensor)
 ```
 Computes the major transpose of a fourth order tensor.
 """
-@generated function majortranspose{dim}(S::FourthOrderTensor{dim})
-    expr = Expr(:tuple)
-    for l in 1:dim, k in 1:dim, j in 1:dim, i in 1:dim
-        push!(expr.args, :(get_data(S)[$(compute_index(get_base(S), k, l, i, j))]))
-    end
-    return quote
-        $(Expr(:meta, :inline))
-        @inbounds return Tensor{4, dim}($expr)
-    end
+@inline function majortranspose{dim}(S::Tensor{4, dim})
```

Comment:
This method should have `S::FourthOrderTensor{dim}` as input but always return `Tensor{4, dim}` so just change the input type signature.